cromulent/model.py

```diff
@@ -743,7 +747,13 @@ def _toJSON(self, done, top=None):
 					d[k] = v._toJSON(done=done, top=top)
 				elif type(v) is list:
 					newl = []
+					uniq = []
```

Comment:
Can/should this be a `set`? And should the values in `uniq` be `ni`, or `id(ni)`?

Reply:
I considered set() but thought that there might be cases when the order is important to preserve, hence the more verbose for loop.

For `ni` vs `id(ni)` ... when would there be two different instances that hash to the same `id()`? (Or am I confused as to the question? :) )

Reply:
The only thing `uniq` is being used for is testing if the value has been seen before, right? If ordering is important, than yes, this is fine. But I didn't see any use where that would matter.

Reply:
Well, if `multiple_instances_per_property` is meant to be about object identity (as opposed to equality), I think `id(ni)` is the way to go, because things like `__eq__` and `__hash__` can be overridden.

Reply:
Use case for order: If you treat the JSON as JSON, then the order of authors in the array could be used as a stand-in for actual ordering of authors in the data. Yes this would only work in JSON-LD rather than any other serialization and wouldn't survive any round-tripping, but JSON-LD is the primary target.   (And repeat for any other ordering you can think of)

Understood about overriding eq and hash. Just because we don't do it doesn't mean some entity that ends up there to be serialized hasn't done it.

Reply:
But the only accesses to `uniq` are membership test `ni in uniq` and adding data `uniq.append(ni)`. The ordering doesn't matter, right?